tests/core/warning.c

```diff
@@ -0,0 +1,49 @@
+#include "clar_libgit2.h"
+#include "warning.h"
+
+static git_warning g_warning = { GIT_WARNING_GENERIC, "Not really an error" };
+static int g_dummy_payload;
+
+void test_core_warning__cleanup(void)
+{
+	git_warning_set_callback(NULL, NULL);
+}
+
+void test_core_warning__zero_on_unset(void)
+{
+	cl_git_pass(git_warning__raise(&g_warning));
+}
+
+static int values_callback(git_warning *warning, void *payload)
+{
+	cl_assert_equal_p(&g_warning, warning);
+	cl_assert_equal_p(&g_dummy_payload, payload);
+
```

Comment:
Shouldn't you set a static value here to assert in the caller that the callback was in fact invoked?